Custom Level Changes

Bletch1971 opened the pull request:
1. Added CSV Import feature for both Player and Dino level customizations.
2. Added CSV Export feature for both Player and Dino level customizations.
3. Added CSV Parse package

Bletch1971 committed:
Custom Level Changes

1. Added Import feature for both Player and Dino level customizations.
2. Added Export feature for both Player and Dino level customizations.
3. Added CSV Parse package

ChronosWS commented on ARK Server Manager/ServerSettingsControl.xaml.cs:
```diff
@@ -548,6 +554,100 @@ private void MaxXPDino_Reset(object sender, RoutedEventArgs e)
+                            switch (action)
+                            {
+                                case ServerSettingsImportExportAction.DinoLevels:
+                                    dialog.Title = "Save level customizations";
```
Need to pull this from a resource file so it can be localized.

ChronosWS replied:
All of these dialogs are very similar except for some text it looks like.  Consider refactoring to a method which generates the dialog from the parameters.

ChronosWS commented on ARK Server Manager/ServerSettingsControl.xaml.cs:
```diff
@@ -548,6 +554,100 @@ private void MaxXPDino_Reset(object sender, RoutedEventArgs e)
+                                    break;
+
+                                case ServerSettingsImportExportAction.PlayerLevels:
+                                    dialog.Title = "Save level customizations";
```
As above.

ChronosWS commented on ARK Server Manager/ServerSettingsControl.xaml.cs:
```diff
@@ -548,6 +554,100 @@ private void MaxXPDino_Reset(object sender, RoutedEventArgs e)
+                        }
+                        catch (Exception ex)
+                        {
+                            MessageBox.Show(ex.Message, "Import Error", MessageBoxButton.OK, MessageBoxImage.Error);
```
Localize dialog title

ChronosWS commented on ARK Server Manager/ServerSettingsControl.xaml.cs:
```diff
@@ -548,6 +554,100 @@ private void MaxXPDino_Reset(object sender, RoutedEventArgs e)
+                            {
+                                case ServerSettingsImportExportAction.DinoLevels:
+                                    dialog.IsFolderPicker = false;
+                                    dialog.Title = "Select level customization CSV file";
```
Localize

ChronosWS commented on ARK Server Manager/ServerSettingsControl.xaml.cs:
```diff
@@ -548,6 +554,100 @@ private void MaxXPDino_Reset(object sender, RoutedEventArgs e)
+
+                                case ServerSettingsImportExportAction.PlayerLevels:
+                                    dialog.IsFolderPicker = false;
+                                    dialog.Title = "Select level customization CSV file";
```
Localize

ChronosWS commented on ARK Server Manager/ServerSettingsControl.xaml.cs:
```diff
@@ -548,6 +554,100 @@ private void MaxXPDino_Reset(object sender, RoutedEventArgs e)
+                            }
+                        }
+                        catch (Exception ex)
+                        {
```
Localize

Bletch1971 commented:
I have added a method to GlobalizedApplication to return a string instead of an object in another pull request. I will wait until that pull request it merged and then make the changes requested above.

Bletch1971 committed:
Custom Level Code Modifications

1. Have localized the strings.
2. Have cut down the code to retreive the dialog and perform the custom
level action.

Bletch1971 commented:
All modifications have been made. 

I HAVE NOT changed all the strings to use the Globalization as it falls outside the scope of the change and I would prefer to do it in a separate pull request.
